src/index.js

```diff
@@ -345,12 +345,30 @@ function getElementsByName (name, context = document) {
   return getMatchingElementsByName(elementIterator, name)
 }
 
+function closest (selector, context) {
```

Comment:
@nolanlawson I tried to use `getFirstMatchingAncestor` but passing in `ast.nodes` didn't work, even though it expects an array of nodes...

Reply:
@tedconn I looked into it, it's because the top-level `ast.nodes` is for a comma-separated list of selectors, e.g. `.foo, .bar, .baz`. So if it's passed directly in to `matches()` then it fails because we never get into the `if` or `else if` blocks, and just return true.